### PR TITLE
Prettier

### PR DESCRIPTION
**Pull request by josojo**
This PR is just a proposal.

Drop a comment, is prettier hot or not?

I would only put more work into it, if you guys like it.

-------

As I am not the cleanest coder, I thought I could also improve our codebase by installing an auto-formating tool for our solidity code.
According to the chainlink folks, prettier is a standard development tool, easily compatible with solhint.

I used the following guide for installing it:
https://medium.com/coinmonks/introduction-to-solidity-linting-and-formatting-e838c074791a

---
testplan:

de-format a *.sol and format it again by running 
```
npm run prettier:solidity
```

### CODE REVIEW

File: contracts/EpochTokenLocker.sol
Comment (josojo) [inline contracts/EpochTokenLocker.sol:84]: Here, the tool is joking us!

File: contracts/EpochTokenLocker.sol
Comment (fleupold) [inline contracts/EpochTokenLocker.sol:16]: I like this change. We should be explicit about width where ever we can

File: contracts/EpochTokenLocker.sol
Comment (fleupold) [reply contracts/EpochTokenLocker.sol:84]: Should probably be an issue in their github?

**Review by fleupold — APPROVED** [review]
Overall, I like this change. The less we have to think about formatting the better in my opinion.

File: .solhint.json
Comment (bh2smith) [inline .solhint.json:14]: This is way to wide! I think this should be reverted back to 129 (which was already quite graceful).

File: contracts/DevDependencies.sol
Comment (bh2smith) [inline contracts/DevDependencies.sol:12]: How are you avoiding this error?

File: contracts/EpochTokenLocker.sol
Comment (bh2smith) [reply contracts/EpochTokenLocker.sol:16]: I am less excited about this, but its alright I guess...

File: contracts/Migrations.sol
Comment (bh2smith) [inline contracts/Migrations.sol:1]: Finally this file doesn't have to be ignored anymore!

File: contracts/libraries/SnappBaseCore.sol
Comment (bh2smith) [inline contracts/libraries/SnappBaseCore.sol:53]: I am not terribly excited about these things

**Review by bh2smith — APPROVED** [review]
Looks pretty good! Some comments and suggestions inline!

File: .solhint.json
Comment (josojo) [reply .solhint.json:14]: Unfortunately, I the prettier is not able to format everything with max-line-length of 129. This would require another issue in their github repo. Hence, I would like to stick with this wide max-line-length in order to use the prettier at all.

File: contracts/DevDependencies.sol
Comment (josojo) [reply contracts/DevDependencies.sol:12]: Setting "no-empty-blocks": "off" in the solhint configure file

Comment (bh2smith) [reply contracts/DevDependencies.sol:12]: OK... Are we sure we want to universally avoid this warning?

File: .solhint.json
Comment (bh2smith) [reply .solhint.json:14]: I guess. although prettier is making our code less than 100 anyway... Seems fine to merge then.

File: contracts/EpochTokenLocker.sol
Comment (josojo) [reply contracts/EpochTokenLocker.sol:84]: I opened an issue over here:
https://github.com/prettier-solidity/prettier-plugin-solidity/issues/205

**Comment by josojo** [comment]
It seems like this PR would be ready to merge. Though the merge conflicts with the other PRs might become a nightmare, hence, I would prefer to merge this PR, once no other PR is open and do the refactoring only in this PR.

This also allows us to wait for an answer on the open issue about the [msg.sender] formatting.

**Comment by bh2smith** [comment]
Closing in favour of more recent version #324

**Pull request closed without merging**
